packages/machine/src/models/state-channel.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  AssetType,
+  ETHBucketAppState,
+  NetworkContext
+} from "@counterfactual/types";
+import { Zero } from "ethers/constants";
+import { INSUFFICIENT_FUNDS } from "ethers/errors";
+import { BigNumber } from "ethers/utils";
+
+import {
+  freeBalanceTerms,
+  getFreeBalanceAppInterface
+} from "../ethereum/utils/free-balance";
+
+import { AppInstance } from "./app-instance";
+
+// TODO: Hmmm this code should probably be somewhere else?
+const HARD_CODED_ASSUMPTIONS = {
+  freeBalanceDefaultTimeout: 100,
+  freeBalanceInitialStateTimeout: 100,
+  appSequenceNumberForFreeBalance: 0
+};
+
+const enum Errors {
+  APPS_NOT_EMPTY = "Expected the appInstances list to be empty but size was nonzero",
+  APP_DOES_NOT_EXIST = "Attempted to edit an appInstance that does not exist",
+  FREE_BALANCE_MISSING = "Cannot find ETH Free Balance App in StateChannel",
+  FREE_BALANCE_IDX_CORRUPT = "Index used to find ETH Free Balance is broken",
+  INSUFFICIENT_FUNDS = "Attempted to install an appInstance without sufficient funds"
+}
+
+function createETHFreeBalance(
+  multisigAddress: string,
+  multisigOwners: string[],
+  ethBucketAddress: string
+) {
+  // Making these values constants to be extremely explicit about
+  // the built-in assumption here.
+  const signingKeyForFreeBalanceForPerson1 = multisigOwners[0];
+  const signingKeyForFreeBalanceForPerson2 = multisigOwners[1];
+
+  return new AppInstance(
+    multisigAddress,
+    multisigOwners,
+    HARD_CODED_ASSUMPTIONS.freeBalanceDefaultTimeout,
+    getFreeBalanceAppInterface(ethBucketAddress),
+    freeBalanceTerms,
+    false,
+    HARD_CODED_ASSUMPTIONS.appSequenceNumberForFreeBalance,
+    {
+      alice: signingKeyForFreeBalanceForPerson1,
+      bob: signingKeyForFreeBalanceForPerson2,
+      aliceBalance: Zero,
+      bobBalance: Zero
+    },
+    0,
+    HARD_CODED_ASSUMPTIONS.freeBalanceInitialStateTimeout
+  );
+}
+
+export class StateChannel {
+  constructor(
+    public readonly multisigAddress: string,
+    public readonly multisigOwners: string[],
+    private readonly appInstances: Readonly<Map<string, AppInstance>> = new Map<
+      string,
+      AppInstance
+    >([]),
+    private readonly freeBalanceAppIndexes: Readonly<
+      Map<AssetType, string>
+    > = new Map<AssetType, string>([]),
+    private readonly monotonicallyIncreasingSeqNo: number = 0
+  ) {}
+
+  public get sequenceNumber() {
+    return this.monotonicallyIncreasingSeqNo;
```

Comment:
IMO the simplest naming scheme is:

- `numActiveApps` - number of active apps - increases when an app is installed, decreases when an app is uninstalled
- `numInstalledApps` - increases when app is installed, does not decrease when app is uninstalled

the second one could also plausibly be called `monotonicNumInstalledApps`